### PR TITLE
[script][circlecheck] Pretty output option

### PR DESCRIPTION
**Pull request by asechrest**
I previously submitted PR #5284 which provided a nicer formatting for `;circlecheck` output. Some folks liked it. Others weren't sure about it. Learned a good lesson to offer new things as an option, rather than forcing it on users.

This PR adds the nicer column formatting as an option, set in your yaml, but defaults to the traditional output via default setting in base.yaml. I also made some changes after seeing some formatting hiccups with other users. Specifically, the output line-wrapped for some users because of the length of the rows. Also, for users who have characters who are high level and have some skills in the sub 200 circle requirements and others beyond, the split formatting wasn't to my liking. Both of these should now be fixed. I've removed superfluous words in the columns that makes them more compact and the output is now substantially smaller width than even the traditional `;circlecheck` output. This should prevent line wrap for most users. And for users who have exceeded the circle 200 requirements, display now uses a simple plus sign to signify the amount over circle 200 reqs, and maintains the column formatting.

The new yaml flag is `circlecheck_prettyprint`. It defaults to false in base.yaml. When false, traditional output is maintained.

NOTE: There is a known issue where the Genie FE double-spaces the output. This is NOT a result of this PR. The un-revised ;circlecheck does the same thing. We're working on understanding why, but most Genie users are utilizing the built in plugin anyway.

Traditional format:

image

When the yaml `circlecheck_prettyprint` is true, new column-formatted output is displayed:

image

Displaying requirements for a target circle well above the current circle:

image

Example output from a character that has some skills sub 200 circle req, and some over it. Note this is on Genie, so it's doublespaced:

image

### CODE REVIEW

**Comment by asechrest** [comment]
Thanks to @MahtraDR for helping test. Also tagging @KatoakDR for review.

File: circlecheck.lic [hunk 1]
```diff
@@ -498,6 +498,9 @@ def main
 
   $debug_mode_cc = args.debug
 
+  # Get settings so we can check for yaml prettyprint true or false
+  @settings = get_settings
```
Comment (rpherbig) [inline circlecheck.lic:502]: If `@settings` is only going to be used on line 516, then I don't think it should be a local var. I instead suggest moving line 516 up here and passing the `prettyprint` value into `display_requirements`.

File: circlecheck.lic [hunk 2]
```diff
@@ -506,37 +509,153 @@ end
 def display_requirements(requirements, level_target, brief)
   can_circle = false
   first_circle_seen = nil
+  column_headings_displayed = false
   spacer_displayed = false
+
+  # Grab the circlecheck_prettyprint yaml setting
+  prettyprint = @settings.circlecheck_prettyprint #Set manually for testing
```
Comment (rpherbig) [inline circlecheck.lic:516]: I'd prefer `pretty_print`

**Comment by rpherbig** [comment]
Code changes look good to me ✔️ 

Let me know if you want more 👀 on this or are ready for it to be merged.

File: circlecheck.lic (hunk 1 above)
Comment (asechrest) [reply circlecheck.lic:502]: Fair, I'll change this.

File: circlecheck.lic (hunk 2 above)
Comment (asechrest) [reply circlecheck.lic:516]: You got it.

**Comment by asechrest** [comment]
> Code changes look good to me ✔️
> 
> Let me know if you want more 👀 on this or are ready for it to be merged.

I think we're good to merge. Thanks for reviewing.

**Comment by asechrest** [comment]
@rpherbig this is ready to merge after the commit I just made. That commit addresses issue #2854 which said ;circlecheck was broken for thieves.

After quite a bit of review, we've determined via preponderance of evidence that Elanthipedia gives the wrong circle requirements for 3rd Lore for Thieves. And because the original author of ;circlecheck used Elanthipedia's numbers, we believe it is also wrong. I've checked against 3 different online circle calculators as well as checking against Genie's exp plugin.

So, my final commit updates the Thief 3rd Lore requirements, and I believe we can close #2854. Let's also close #5285 due to no further response from OP and unable to recreate the issue.

This should be ready to merge.

**Comment by asechrest** [comment]
Resolves #2854 
Resolves #5285

**Pull request merged**